### ISSUE TITLE
Add up-front explanation of how Max-Message-Size, block transfer, and BERT fit together

### ISSUE THREAD

**Issue by cabo** [core-wg/coap-tcp-tls#160]
Adam notes:

> Perhaps an explanation closer to the front that ties Max-Message-Size, block transfer, and the BERT mechanism together would be useful.

**Comment by cabo** [comment]
This also could include a discussion how setting Max-Message-Size below 1152 could potentially be used to increase performance, but this should not cause errors.

**Comment by smolboarm** [comment]
The one problem with Max-Message-Size and BERT is next:
Packet Length (first field of header) includes length of Options and Payload length. So with BERT enabled there is no clear algorithm how to decide (maximum) BERT block size.
Example: 
Max-Message-Size = 10k
BERT sub-block size = 1k (by spec).

If we just divide 10k/1k then it looks like we could have 10 BERT sub-blocks in the one CoAP packet payload. But if packet has Options, which is also included into packet length - then we will not have space for options (whole 10k used for payload).

Also Max-Message-Size not clearly defined: is it message size including CoAP header or it does not include CoAP header and affects only length field (i.e. affects only Options + payload marker + payload size).

Currently define maximum BERT block payload as: 
int maxBertBlockSize = (Max-Message-Size / 1024) - 1;
if maxMessageSize > 2k, else it is always 1k (1152 ... 2048 bytes)

This is needed to have space for options in CoAP packet.

Basically, my opinion, that BERT block transfers should be enabled only if Max-Message-Size is greater than 2560 bytes (2.5k), because below 2k it is reduced to 1k blocks which is same like usual block transfers blocks. And maximum BERT block size should be always (Max-Message-Size / 1024) - 1.

**Comment by cabo** [comment]
@smolboarm: would you agree that eaaaf3b addresses your comment?  (See also #147.)
